homeassistant/components/hydrawise/__init__.py

```diff
@@ -50,8 +53,16 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
 
 async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
     """Set up Hydrawise from a config entry."""
-    access_token = config_entry.data[CONF_API_KEY]
-    hydrawise = legacy.LegacyHydrawiseAsync(access_token)
+
+    if config_entry.data[CONF_USERNAME] == "" or config_entry.data[CONF_PASSWORD] == "":
```

Comment:
I'd rather see `if CONF_USERNAME in config_entry.data:` instead of checking empty strings

Reply:
Fixed.